app/services/parameters/caster.rb

```diff
@@ -0,0 +1,120 @@
+module Foreman
+  module Parameters
+    class Caster
+      attr_reader :value
+
+      def initialize(item, options = {})
+        defaults = {
+          :attribute_name => :value,
+          :to => :string
+        }
+        options.reverse_merge!(defaults)
+        @item, @options = item, options
+        @value = @options[:value] || @item.send(@options[:attribute_name])
+      end
+
+      def cast!
+        @item.send("#{@options[:attribute_name]}=", casted_value)
+      end
+
+      def cast
+        casted_value
+      end
+
+      private
+
+      def casted_value
+        case @options[:to]
+        when :string
+          cast_string
+        when :integer
+          cast_integer
+        when :real
+          cast_real
+        when :boolean
+          cast_boolean
+        when :array
+          cast_array
+        when :hash
+          cast_hash
+        when :json
+          cast_json
+        when :yaml
+          cast_yaml
+        else
+          #cowardly refusing to cast
+          value
+        end
+      end
+
+      def cast_string
+        value.to_s
+      end
+
+      def cast_boolean
+        return true if Foreman::Cast.to_bool(value)
+        return false if Foreman::Cast.to_bool(value) == false
```

Comment:
Avoid running this twice - you could return the return value unless it's nil, or assign it to a local variable.